components/src/dynamo/common/utils/video_utils.py

```
@@ -242,3 +245,297 @@ def encode_to_video_bytes(
+        cmd += ["-f", container, output_path]
+
+        logger.info(
+            "Encoding %d frames (%dx%d @ %d fps) via %s on %s",
+            num_frames,
+            width,
+            height,
+            fps,
+            encoder,
+            device if hw_accel == "xpu" else hw_accel,
+        )
+        proc = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)